src/utils/web.py

```diff
@@ -143,7 +166,19 @@ def getUrlFd(url, headers=None, data=None, timeout=None):
         else:
             request = url
             request.add_data(data)
-        fd = urlopen(request, timeout=timeout)
+        if context is not None and ca_file is False and nocert is True:
+            fd = urlopen(request, timeout=timeout, context=context)
+        elif context is not None and ca_file is False and nocert is False:
+            fd = urlopen(request, timeout=timeout, context=context)
+        elif context is not None and ca_file is True and nocert is True:
+            fd = urlopen(request, timeout=timeout, context=context)
+        elif context is not None and ca_file is True and nocert is False:
+            try:
+                fd = urlopen(request, timeout=timeout)
+            except:
+                fd = urlopen(request, timeout=timeout, context=context)
+        else:
+            fd = urlopen(request, timeout=timeout)
```

Comment:
I see a lot of duplication in this block; what are all the cases you are trying to handle?